pubtools/pulplib/_impl/model/maintenance.py

```diff
@@ -0,0 +1,200 @@
+import logging
+import datetime
+import os
+
+import jsonschema
+
+from pubtools.pulplib._impl import compat_attr as attr
+from ..schema import load_schema
+from .common import InvalidDataException
+
+
+LOG = logging.getLogger("pubtools.pulplib")
+
+
+USER = os.environ.get("USER")
+HOSTNAME = os.environ.get("HOSTNAME")
+
+
+def iso_time_now():
+    return datetime.datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
+
+
+@attr.s(kw_only=True, frozen=True)
+class MaintenanceEntry(object):
+    """Details about the maintenance status of a specific repository."""
+
+    repo_id = attr.ib(type=str)
+    """ID of repository in maintenance.
+    Note: there is no guarantee that a repository of this ID currenly exists
+          in the Pulp server."""
+    message = attr.ib(default=None, type=str)
+    """Why this repository is in maintenance."""
+    owner = attr.ib(default=None, type=str)
+    """Who set this repository in maintenance mode."""
+    started = attr.ib(default=iso_time_now(), type=datetime.datetime)
+    """:class:`~datetime.datetime` in UTC at when the maintenance started."""
+
+
+@attr.s(kw_only=True, frozen=True)
+class MaintenanceReport(object):
+    """Represents the maintenance status of Pulp repositories.
+
+    On release-engineering Pulp servers, it's possible to put individual repositories
+    into "maintenance mode".  When in maintenance mode, external publishes of a repository
+    will be blocked.  Other operations remain possible.
+
+    This object holds information on the set of repositories currently in maintenance mode.
+    """
+
+    _OWNER = "%s@%s" % (USER, HOSTNAME) if all([USER, HOSTNAME]) else "pubtools.pulplib"
+
+    _SCHEMA = load_schema("maintenance")
+
+    last_updated = attr.ib(default=iso_time_now(), type=datetime.datetime)
+    """:class:`~datetime.datetime` in UTC when this report was last updated,
+    if it's the first time the report is created, current time is used."""
+
+    last_updated_by = attr.ib(default=None, type=str)
+    """Person/party who updated the report last time."""
+
+    entries = attr.ib(default=attr.Factory(tuple), type=tuple)
+    """A tuple of :class:`MaintenanceEntry` objects, indicating
+    which repositories are in maintenance mode and details.
+    If empty, then it means no repositories are in maintenance mode.
+    """
+
+    @entries.validator
+    def check_duplicates(self, attribute, value):
```

Comment:
This should be named with _ as well as it's not public API.

Also, for the pylint issue, personally I prefer the style of naming the argument as simply _.  Although it does have the downside that someone has to read the validator docs to understand what that unused arg is.

```suggestion
    def _check_duplicates(self, _, value):
```